web-api/migration-terraform/main/lambdas/migration.js

```diff
@@ -1,41 +1,55 @@
 const AWS = require('aws-sdk');
 const { get } = require('lodash');
+const { migrateRecords: migrations } = require('./migration-segments');
 
 const dynamodb = new AWS.DynamoDB({
   maxRetries: 10,
   retryDelayOptions: { base: 300 },
 });
 
-const documentClient = new AWS.DynamoDB.DocumentClient({
+const docClient = new AWS.DynamoDB.DocumentClient({
   endpoint: 'dynamodb.us-east-1.amazonaws.com',
   region: 'us-east-1',
   service: dynamodb,
 });
 
-const processItems = async items => {
+const processItems = async ({ documentClient, items, migrateRecords }) => {
   const promises = [];
+
+  items = await migrateRecords({ documentClient, items });
+
   for (const item of items) {
-    // TODO: your migration code would go here
-    const result = documentClient
-      .put({
-        Item: item,
-        TableName: process.env.DESTINATION_TABLE,
-      })
-      .promise();
-    promises.push(result);
+    promises.push(
+      documentClient
+        .put({
+          Item: item,
+          TableName: process.env.DESTINATION_TABLE,
+        })
+        .promise(),
+    );
```

Comment:
What happens if one of these fails due to throttling? Would it just error out, and then the item would still be in the DDB Stream to be retried?

Reply:
From what I know, Kinesis doesn't move onto the next batch of events unless it processes everything successfully.  It will keep retrying the batch.